agent/structs/config_entry_file_system_certificate.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package structs
+
+import (
+	"github.com/hashicorp/consul/acl"
+)
+
+// FileSystemCertificateConfigEntry manages the configuration for a certificate
+// and private key located in the local file system.
+type FileSystemCertificateConfigEntry struct {
+	// Kind of config entry. This will be set to structs.FileSystemCertificate.
+	Kind string
+
+	// Name is used to match the config entry with its associated file system certificate.
+	Name string
+
+	// Certificate is the optional path to a client certificate to use for TLS connections.
+	Certificate string
+
+	// PrivateKey is the optional path to a private key to use for TLS connections.
+	PrivateKey string
+
+	Meta               map[string]string `json:",omitempty"`
+	Hash               uint64            `json:",omitempty" hash:"ignore"`
+	acl.EnterpriseMeta `hcl:",squash" mapstructure:",squash"`
```

Comment:
should this be a `*acl.EnterpriseMeta`? i feel like I've seen inconsistent usage of it that way so either probably works

Reply:
This is shaped the same way as `InlineCertificateConfigEntry` today, so it should work. Can verify that things work with namespaces enabled before merge

https://github.com/hashicorp/consul/blob/f2b26ac1949e0412c2c7afb80f7c514a29464c38/agent/structs/config_entry_inline_certificate.go#L21-L37